### PR TITLE
Remove bluebird dependency

### LINKED ISSUE #5: Remove bluebird from dependencies
ES6 includes a Promise implementation and even though bluebird one is nice, this library doesn't really need it at all.

### PR DESCRIPTION
**Pull request by ruiaraujo**
I think this version is equivalent to the Bluebird implementation.
Closes #5

### CODE REVIEW

File: modules/helpers/utils.js [hunk 1]
```diff
@@ -45,6 +44,22 @@ export function filterAndFlattenComponents(components) {
   return flattened;
 }
 
+function promiseMapSeries(array, iterator, thisArg) {
+  const length = array.length;
+  let current = Promise.resolve();
+  const results = new Array(length);
+  const cb = !!thisArg ? iterator.bind(thisArg) : iterator;
```
Comment (AVVS) [inline modules/helpers/utils.js]: this is slow, cant use `.bind`

File: modules/helpers/utils.js [hunk 2]
```diff
@@ -45,6 +44,22 @@ export function filterAndFlattenComponents(components) {
...
+
+  for (let i = 0; i < length; ++i) {
+    current = results[i] = current.then(function promiseCreator(index) {
```
Comment (AVVS) [inline modules/helpers/utils.js]: creating functions in a loop is also slow, and then binding them is even worse

**Comment by AVVS** [comment]
Maybe we can do the polyfills properly?

**Comment by ruiaraujo** [comment]
I changed it to allow using the bluebird implementation for the mapSeries. As for the rest of the changes is basically to rely on the subset of the Promise spec instead on the features that bluebird provides.

**Comment by AVVS** [comment]
your changes were actually breaking the desired flow of promises - making it parallel instead of series as intended, I've coded a solution in #10, please take a look if this solves your cases

**Comment by ruiaraujo** [comment]
@AVVS I don't think so but it doesn't matter since you ended up doing it yourself and so I can kill my fork. ;-)

**Comment by AVVS** [comment]
I did, but here is the explanation:

```diff
+  const cb = (arr, index) => () => iterator(array[index], index, arr);
+
+  for (let i = 0; i < length; ++i) {
-    // create array of functions, which would invoke iterator when called
-    // it's done in a sync fashion, but it's still fine
+    current = results[i] = current.then(cb(array, i));
+  }
+
-  // now we launch all of them in parallel, effectively create a simple async map,
-  // to create mapSeries we need to do recursive walk over results, func 1 -> complete -> func 2 -> etc
+  return Promise.all(results);
```

**Comment by ruiaraujo** [comment]
I realize that and I see the difference of the implementation, but considering the practical way that it gets used at his moment they are equivalent IMO. However that doesn't matter and thanks for being openminded.

**Pull request closed without merging**